src/calendar/__specs__/month_spec.js

```diff
@@ -220,6 +220,64 @@ describe('Month', () => {
     })
   })
 
+  context('in "range" and "selectedDaysLimit" mode', () => {
+    let onChange, month
+    beforeEach(() => {
+      onChange = sinon.spy()
+      month = render({
+        activeMonth: new Date(2015, 5, 1),
+        selectionMode: 'range',
+        onChange,
+        selectedDaysLimit: 5
+      })
+    })
+
+    it('can select single day range', () => {
+      clickOnDays(month, 5, 5)
+      assert(onChange.calledWith({
+        start: new Date(2015, 5, 6),
+        end: new Date(2015, 5, 6),
+        inProgress: false
+      }))
+    })
+
+    it('can select days range if its in the Limit range', () => {
+      clickOnDays(month, 5, 9)
+      assert(onChange.calledWith({
+        start: new Date(2015, 5, 6),
+        end: new Date(2015, 5, 10),
+        inProgress: false
+      }))
+    })
+
+    it('can select days range but it will be adjusted according to the Limit', () => {
+      clickOnDays(month, 5, 16)
+      assert(onChange.calledWith({
+        start: new Date(2015, 5, 6),
+        end: new Date(2015, 5, 11),
+        inProgress: false
+      }))
+    })
+
+    it('can select days range in inverted order if its in the Limit range', () => {
+      clickOnDays(month, 9, 5)
+      assert(onChange.calledWith({
+        start: new Date(2015, 5, 6),
+        end: new Date(2015, 5, 10),
+        inProgress: false
+      }))
+    })
+
+    it('can select days range in inverted order but it will be adjusted according to the Limit', () => {
+      clickOnDays(month, 16, 5)
+      assert(onChange.calledWith({
+        start: new Date(2015, 5, 6),
+        end: new Date(2015, 5, 11),
+        inProgress: false
+      }))
+    })
```

Comment:
I think it's better to specify a few contexts like

- "when the first date is before the second one"
- "when the first date is after the second one"
- etc